File: test_api.js
const axios = require('axios');

const BASE_URL = 'http://127.0.0.1:59212/api';
let token = '';

async function test() {
  try {
    console.log('=== 1. Login ===');
    const loginRes = await axios.post(`${BASE_URL}/auth/admin-login`, {
      username: 'admin',
      password: 'admin123'
    });
    token = loginRes.data.data.token;
    console.log('Token obtained');

    const headers = { Authorization: `Bearer ${token}` };

    console.log('\n=== 2. Dashboard ===');
    const dashRes = await axios.get(`${BASE_URL}/admin/dashboard`, { headers });
    console.log(JSON.stringify(dashRes.data, null, 2).substring(0, 3000));

    console.log('\n=== 3. Orders ===');
    const ordersRes = await axios.get(`${BASE_URL}/admin/orders?page=1&pageSize=10`, { headers });
    console.log(JSON.stringify(ordersRes.data, null, 2).substring(0, 2000));

    console.log('\n=== 4. Suppliers ===');
    const suppRes = await axios.get(`${BASE_URL}/admin/suppliers`, { headers });
    console.log(JSON.stringify(suppRes.data, null, 2).substring(0, 2000));

    console.log('\n=== 5. Card Pool Stats ===');
    const cardRes = await axios.get(`${BASE_URL}/admin/card-pool/stats`, { headers });
    console.log(JSON.stringify(cardRes.data, null, 2));

    console.log('\n=== 6. Risk Logs ===');
    const riskRes = await axios.get(`${BASE_URL}/admin/risk/logs?limit=10`, { headers });
    console.log(JSON.stringify(riskRes.data, null, 2).substring(0, 2000));

  } catch (e) {
    console.error('Error:', e.response?.data || e.message);
  }
}

test();

File: _test_api.py
#!/usr/bin/env python3
import requests, json

B = 'http://127.0.0.1:59212/api'

r = requests.post(B + '/auth/login', json={'phone': '13800138001', 'password': '123456'})
token = r.json()['data']['token']
headers = {'Authorization': 'Bearer ' + token}

print('=== 测试 /admin/products ===')
r = requests.get(B + '/admin/products?page=1&pageSize=1', headers=headers)
data = r.json()
print(f'success: {data.get("success")}')
if data.get('success'):
    p = data['data']['list'][0]
    print(f'商品: {p["name"]}')
    print(f'  channelCount: {p.get("channelCount")}')
    print(f'  activeChannelCount: {p.get("activeChannelCount")}')
    print(f'  channelStatus: {p.get("channelStatus")}')
    print(f'  supplier_count: {p.get("supplier_count")}')
    print(f'  hasFallback: {p.get("hasFallback")}')
    print(f'  region_limited: {p.get("region_limited")}')
    print(f'  available_regions: {p.get("available_regions")}')
    print(f'  sync_batch: {p.get("sync_batch")}')
    print(f'  stock_sync_history: {p.get("stock_sync_history")}')
    print(f'  suppliers count: {len(p.get("suppliers", []))}')
    if p.get('suppliers'):
        for s in p['suppliers']:
            print(f'    - {s["name"]} (main={s["is_main"]}, channels={s["channel_count"]})')
